sheets/api.py

```diff
@@ -463,21 +521,69 @@ def create_coupon_assignment_sheet(self, coupon_req_row):
 
         return bulk_coupon_sheet
 
-    def write_results_to_sheets(self, processed_requests):
+    def create_and_update_sheets(self, processed_requests):
         """
         Updates the coupon request sheet and creates a coupon assignment sheet for every successfully
         processed coupon request.
 
         Args:
             processed_requests (list of ProcessedRequest): List of ProcessedRequest objects
+
+        Returns:
+            list of pygsheets.Spreadsheet: Spreadsheet objects representing the spreadsheets that were
+                created
         """
-        if not processed_requests:
-            return
-        # Set checkboxes
+        # Set processed dates
         self.update_coupon_request_processed_dates(processed_requests)
-        # Write new sheets with codes
-        for processed_request in processed_requests:
+        # Write new assignment sheets with coupon codes
+        return [
             self.create_coupon_assignment_sheet(processed_request.coupon_req_row)
+            for processed_request in processed_requests
+        ]
+
+    def process_sheet(self, limit_row_index=None):
```

Comment:
It doesn't look like this function is covered at all by tests. Can we add tests for this function? https://codecov.io/gh/mitodl/mitxpro/pull/1349/src/sheets/api.py#L544